### PR TITLE
Add Crbug components to chromestatus features, and allow users to file bugs from chromestatus.com

### LINKED ISSUE #297: Offer better bug filing affordances from chromestatus entries
Before a feature is launched, provide a link that goes to a new bug prepopulated with blocking:<feature bug>

After a feature is launched, provide a link that goes to a new bug prepopulated with the right component.

### PR DESCRIPTION
**Pull request by clelland**
This change adds a bug_components field to the feature model, and exposes a "File new bug" button in the feature list. If the components are present, they will be auto-filled in the new bug form on crbug. If the feature has not yet launched, and has a launch bug number, then the new bug will be created as a blocker by default.

Fixes #297

### CODE REVIEW

File: static/elements/chromedash-feature.html [hunk 1]
```diff
@@ -89,6 +89,13 @@
             <iron-icon icon="icons:extension" class="experimental"></iron-icon>
           </span>
         </template>
+        <template is="dom-if" if="{{feature.bug_components}}">
+          <span class="tooltip" title="File a bug">
```
Comment (ebidel) [inline static/elements/chromedash-feature.html]: "File a bug against this feature"?

File: static/elements/chromedash-feature.html [hunk 2]
```diff
@@ -381,18 +388,46 @@
...
+        return;
       }
       var m = link.match(/[\/|?id=]([0-9]+)$/); // pull out bug number.
       if (m) {
-        return '#' + m[1];
+        return m[1];
+      }
+    },
+    _computeCrbug: function(link) {
+      var CrBugNumber = this._computeCrbugNumber(link)
```
Comment (ebidel) [inline static/elements/chromedash-feature.html]: crBugNumber

File: static/elements/chromedash-feature.html [hunk 3]
```diff
@@ -381,18 +388,46 @@
...
+      if (CrBugNumber) {
+        return '#' + CrBugNumber;
       }
       return link;
+    },
+    _computeNewBugUrl: function(feature) {
+      var url = "https://bugs.chromium.org/p/chromium/issues/entry";
```
Comment (ebidel) [inline static/elements/chromedash-feature.html]: single quotes for js strings

File: static/elements/chromedash-feature.html [hunk 4]
```diff
@@ -381,18 +388,46 @@
...
+      var params = [];
+      if (feature.bug_components.length) {
+        params.push("components=" + feature.bug_components.map(encodeURIComponent));
+      }
+      var crbugNumber = this._computeCrbugNumber(feature.bug_url)
```
Comment (ebidel) [inline static/elements/chromedash-feature.html]: camel casing: `crBugNumber`

File: static/elements/chromedash-feature.html [hunk 5]
```diff
@@ -381,18 +388,46 @@
...
+      if (crbugNumber && this._computePreLaunch(feature.impl_status_chrome)) {
```
Comment (RByers) [inline static/elements/chromedash-feature.html]: Also cc the feature owners?

File: static/elements/chromedash-feature.html [hunk 6]
```diff
@@ -381,18 +388,46 @@
...
+        params.push("blocking="+crbugNumber);
```
Comment (ebidel) [inline static/elements/chromedash-feature.html]: spacing: `"'blocking=" + crbugNumber`

File: static/elements/chromedash-feature.html [hunk 7]
```diff
@@ -381,18 +388,46 @@
...
+      if (CrBugNumber) {
+        return '#' + CrBugNumber;
       }
       return link;
+    },
+    _computeNewBugUrl: function(feature) {
+      var url = "https://bugs.chromium.org/p/chromium/issues/entry";
+      var params = [];
+      if (feature.bug_components.length) {
+        params.push("components=" + feature.bug_components.map(encodeURIComponent));
```
Comment (RByers) [inline static/elements/chromedash-feature.html]: Suggest using just `Blink` if no components are supplied - such bugs should be triaged quicker (one less queue to go through).

**Comment by ebidel** [comment]
Is there any way to bring in the list of "Blink-" component names dynamically on new feature form page? I'm worried people will create typos or the labels will get out of sync with monorail.

**Comment by RByers** [comment]
> Is there any way to bring in the list of "Blink-" component names dynamically on new feature form page? I'm worried people will create typos or the labels will get out of sync with monorail.

I agree a free form field is problematic.  Let's discuss on #297 - I just had wrote another idea to improve this there.

File: static/elements/chromedash-feature.html (hunk 6 above)
Comment (clelland) [inline static/elements/chromedash-feature.html]: Done.

File: static/elements/chromedash-feature.html (hunk 5 above)
Comment (clelland) [inline static/elements/chromedash-feature.html]: Done now, thanks.

File: static/elements/chromedash-feature.html (hunk 4 above)
Comment (clelland) [inline static/elements/chromedash-feature.html]: Changed to crbugNumber for consistentCamelCase. (bug was lowercase originally, as if 'crbug' is considered a single word)

File: static/elements/chromedash-feature.html (hunk 3 above)
Comment (clelland) [inline static/elements/chromedash-feature.html]: All fixed. Thanks.

File: static/elements/chromedash-feature.html (hunk 2 above)
Comment (clelland) [inline static/elements/chromedash-feature.html]: Done

**Comment by clelland** [comment]
I'm splitting this into two -- the bug link code has been moved to #328, and I'll rebase this branch against that one.

**Comment by ebidel** [comment]
@clelland is this ready to look at again? It'll need a rebase.

**Comment by jrobbins** [comment]
This seems obsolete.  It looks like half of it landed years ago.

**Pull request closed without merging**